pkg/utils/release/controller.go

```diff
@@ -109,29 +114,146 @@ func (s *SuiteController) GetPipelineRunInNamespace(namespace, releaseName, rele
 	return nil, err
 }
 
-// GetRelease returns the release with the given name in the given namespace.
-func (s *SuiteController) GetRelease(releaseName, releaseNamespace string) (*v1alpha1.Release, error) {
-	release := &v1alpha1.Release{}
+// CreateReleasePlan creates a new ReleasePlan using the given parameters.
+func (s *SuiteController) CreateReleasePlan(name, source_namespace, application, target_namespace, AutoReleaseLabel string) (*appstudiov1alpha1.ReleasePlan, error) {
+	var releasePlan *appstudiov1alpha1.ReleasePlan
+
+	if AutoReleaseLabel != "" { // AutoReleaseLabel in ReleasePlan is not missing
```

Comment:
I think there is a cleaner way to do this without duplicating so much code. You can probably define the `ReleasePlan` then have your `if` and do like `ReleasePlan.ObjectMeta.Labels[appstudiov1apha1.AutoReleaeLabel] = AutoReleaseLabel`. That may not be 100% correct, but I think it is close. Try it out in your IDE

Reply:
Additionally, why do the release service utils even need the ability to create a `ReleasePlan` with or without the label? Only integration service cares about the label on `ReleasePlan`s. Is it so they can import and use this function in their own tests? If not, I don't see what we gain from this functionality

Reply:
I add this AutoReleaseLabel support for this function for the purpose of completeness (for our usages we shall always use AutoReleaseLabel="" meaning we don't add this label to the CRD

Reply:
1. I created the ReleasePlan CRD for the purpose of completeness to simulate the happy-path with source related ReleaseLink.
2. I don't currently know what will happen if I don't create the ReleasePlan - that should be another Jira ticket
3. I define this CRD to show exactly if adding explicitly the label or not

Reply:
I am not saying we don't need the ReleasePlan. I am saying we don't need it with and without the label. Either is just fine since we don't test based on the label

Reply:
In the function yes to get as an argument what to set the label, but in this specific e2e test we want the creation of ReleasePlan to be on default, means the label should not be set and kept on default

Reply:
I like the go code much better here now, but still not sure why we need the release test functions to care about the label when creating ReleasePlans when the release service does not care about this label

Reply:
Added it to make the path to the happy-path tests easier